src/linker/pipeline.py

```diff
@@ -62,41 +63,47 @@ def get_step_id(self, implementation: Implementation) -> str:
         step_number = str(idx + 1).zfill(len(str(len(self.implementations))))
         return f"{step_number}_{implementation.step_name}"
 
-    def get_input_files(self, implementation: Implementation, results_dir: Path) -> list:
+    def get_input_files(self, implementation: Implementation) -> list:
         idx = self.implementation_indices[implementation.name]
         if idx == 0:
             return [str(file) for file in self.config.input_data]
         else:
-            previous_output_dir = self.get_output_dir(
-                self.implementations[idx - 1], results_dir
-            )
+            previous_output_dir = self.get_output_dir(self.implementations[idx - 1])
             return [str(previous_output_dir / "result.parquet")]
 
-    def get_output_dir(self, implementation: Implementation, results_dir: Path) -> Path:
+    def get_output_dir(self, implementation: Implementation) -> Path:
         idx = self.implementation_indices[implementation.name]
         if idx == len(self.implementations) - 1:
-            return results_dir
+            return self.config.results_dir
 
-        return results_dir / "intermediate" / self.get_step_id(implementation)
+        return self.config.results_dir / "intermediate" / self.get_step_id(implementation)
 
-    def get_diagnostics_dir(self, implementation: Implementation, results_dir: Path) -> Path:
-        return results_dir / "diagnostics" / self.get_step_id(implementation)
+    def get_diagnostics_dir(self, implementation: Implementation) -> Path:
+        return self.config.results_dir / "diagnostics" / self.get_step_id(implementation)
 
-    def build_snakefile(self, results_dir: Path) -> Path:
-        self.write_imports(results_dir)
-        self.write_target_rules(results_dir)
+    @property
+    def snakefile_path(self) -> Path:
+        return self.config.results_dir / "Snakefile"
+
+    def build_snakefile(self) -> Path:
+        if self.snakefile_path.is_file():
+            logger.warning("Snakefile already exists, overwriting.")
```

Comment:
Is this the proper thing to do? Or should we raise?

Reply:
If we run linker in the working directory (i.e. without a timestamped directory) I'd prefer the user to not have to  manually delete the file before running again. I guess we could give them the option to say Y/N to overwrite the directory's contents first